cmsl1t/utils/root_glob.py

```diff
@@ -0,0 +1,75 @@
+"""
+Reproduce the standard glob package behaviour but use TSystem to be able to
+query remote file systems, like xrootd
+"""
+from __future__ import print_function
+from rootpy.ROOT import gSystem
+import glob as gl
+import os.path
+import fnmatch
+
+
+def __directory_iter(directory):
+    while True:
+        file = gSystem.GetDirEntry(directory)
+        if not file:
+            break
+        yield file
+    return
+
+
+def glob(pathname):
+    # Let normal python glob try first
+    try_glob = gl.glob(pathname)
+    if try_glob:
+        return try_glob
+
+    # If pathname does not contain a wildcard:
+    if not gl.has_magic(pathname):
+        return [pathname]
+
+    # Split the pathname into a directory and basename
+    # (which should include the wild-card)
+    dirname, basename = os.path.split(pathname)
+
+    # Uses `TSystem` to open the directory.
+    # TSystem itself wraps up the calls needed to query xrootd.
+    dirname = gSystem.ExpandPathName(dirname)
+    directory = gSystem.OpenDirectory(dirname)
+
+    files = []
+    if directory:
+        for file in __directory_iter(directory):
+            if file in [".", ".."]:
+                continue
+            if not fnmatch.fnmatchcase(file, basename):
+                continue
+            files.append(os.path.join(dirname, file))
+        gSystem.FreeDirectory(directory)
+    return files
+
+
+def iglob(pathname):
+    for name in glob(pathname):
+        yield name
+
+
+if __name__ == "__main__":
```

Comment:
I can do.  How do I unit-test files on an xrootd server, though?  Does the build system have access to this?  I'm not sure how I can fake such a thing either.